packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -534,9 +533,25 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
         address prevADDRESS = messageContext.ovmADDRESS;
         messageContext.ovmADDRESS = eoa;
 
-        // Now actually create the account and get its bytecode. We're not worried about reverts
-        // (other than out of gas, which we can't capture anyway) because this contract is trusted.
-        OVM_ProxyEOA proxyEOA = new OVM_ProxyEOA(0x4200000000000000000000000000000000000003);
+        // Creates a duplicate of the OVM_ProxyEOA located at 0x42....09. Uses the following
+        // "magic" prefix to deploy an exact copy of the code:
+        // PUSH1 0x0D   # size of this prefix in bytes
+        // CODESIZE
+        // SUB          # subtract prefix size from codesize 
+        // DUP1
+        // PUSH1 0x0D
+        // PUSH1 0x00
+        // CODECOPY     # copy everything after prefix into memory at pos 0
+        // PUSH1 0x00
+        // RETURN       # return the copied code
+        address proxyEOA = Lib_EthUtils.createContract(abi.encodePacked(
+            hex"600D380380600D6000396000f3",
```

Comment:
This bytestring looks good; it would also be super obvious from integration tests if it weren't looking.  Since we do not have constructors, then this code could obviously also be put in place without a constructor.  I assume we are going this way for ease of implementation in geth, right @smartcontracts? if so, we could use the `ovmSETCODE` upgrade path in the future.

Reply:
Good call on using SETCODE for this in the future